### PR TITLE
MOS: Avoid commutating operands when doing so would lead to less trivial copies.

### PR DESCRIPTION
**Pull request by asiekierka**
Fixes https://github.com/llvm-mos/llvm-mos/issues/383 .

I can't help but wonder if this case couldn't be caught in a more generalized manner - but I lack the wisdom to get there myself :smiling_face_with_tear:

### CODE REVIEW

File: llvm/lib/Target/MOS/MOSInstrInfo.cpp [hunk 1]
```diff
@@ -267,6 +267,66 @@ bool MOSInstrInfo::findCommutedOpIndices(const MachineInstr &MI,
   return true;
 }
 
+bool MOSInstrInfo::hasCommutePreference(MachineInstr &MI,
```
Comment (mysterymath) [inline llvm/lib/Target/MOS/MOSInstrInfo.cpp:270]: This definitely needs some tests that run scenarios through TwoAddressInstruction. It should be fairly straightforward to get it to commute based on the logic here, but if not, then even a simple regression test captured from the original IR would be useful.

**Review by mysterymath — COMMENTED** [review]
Nice find on the interface for this; another X86-specific gem.

File: llvm/lib/Target/MOS/MOSInstrInfo.cpp (hunk 1 above)
Comment (asiekierka) [reply llvm/lib/Target/MOS/MOSInstrInfo.cpp:270]: We already have general tests for TwoAddressInstruction (`commute.mir`). I added a regression test to the set from the relevant issue's IR.

**Review by mysterymath — APPROVED** [review]
LGTM!

**Pull request merged**